cmd/verify/verify.go

```diff
@@ -40,40 +55,110 @@ func main() {
 	forever := make(chan bool)
 
 	log.Info("starting verify service")
-	var message Message
-	// var header []byte
 
 	go func() {
 		for d := range broker.GetMessages(mq, config.Broker.Queue) {
 			log.Debugf("received a message: %s", d.Body)
-			err := json.Unmarshal(bytes.Replace(d.Body, []byte(`'`), []byte(`"`), -1), &message)
-			if err != nil {
+			var m Message
+			e := json.Unmarshal(bytes.Replace(d.Body, []byte(`'`), []byte(`"`), -1), &m)
+			if e != nil {
 				log.Errorf("Not a json message: %s", err)
+				// Nack errorus message so the server gets notified that something is wrong but don't requeue the message
+				d.Nack(false, false)
+				// Send the errorus message to an error queue so it can be analyzed.
+				if err := broker.SendMessage(mq, d.CorrelationId, config.Broker.Exchange, config.Broker.RoutingError, d.Body); err != nil {
+					log.Error("faild to publish message, reason: ", err)
+				}
+				// Restart on new message
+				continue
+			} else {
+				e := reflect.ValueOf(&m).Elem()
```

Comment:
I'm not sure what is this reflection magic for... Are you validating unmarshalled JSON like that? If struct-fields are not pointers and they are missing in the JSON, unmarshalling will fail anyway. Also, it's better to validate the JSON message against the JSON schema before unmarshalling, IMHO (if you want to have validation at all).

Reply:
That is to ensure that the incoming message doesn't contains null values for the keys we require.
 {"user": ""} or {"user": null} are both valid json and the unmarshaling will not catch that.

Once we have the proper message definitions we could use schema validation but for now this is it.

Reply:
If it's `""`, then it's a valid string - just empty. And we can check if it's empty without reflection. But are you sure that `null`s will not fail during unmarshalling? I think it will if this string is not a pointer in the struct. But I will double-check.

Reply:
Running the unmarshalling on the second example above does not generate an error.